coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -1445,6 +1447,23 @@ func (s *server) CompleteJob(ctx context.Context, completed *proto.CompletedJob)
 				return xerrors.Errorf("calculate auto stop: %w", err)
 			}
 
+			if workspace.AutostartSchedule.Valid {
+				templateScheduleOptions, err := templateScheduleStore.Get(ctx, db, workspace.TemplateID)
+				if err != nil {
+					return xerrors.Errorf("get template schedule options: %w", err)
+				}
+
+				nextStartAt, _ := schedule.NextAutostart(now, workspace.AutostartSchedule.String, templateScheduleOptions)
```

Comment:
I don't think you can ignore the second return value ("allowed") here?

Reply:
Looking at it again maybe I slightly misunderstood it the first time round. 

So looking at it: `NextAutostart` returns the next autostart time after `at` (in this case the value `now`), and returns if this lands on a valid day of the week.

I guess this could cause false-positives to occur on days where autostart is not valid? 

Maybe more important to note that I've just spotted whilst re-reading it: the time returned by `NextAutostart` is localised to the schedules timezone instead of UTC. I think we will probably need to `.UTC()` on the returned time.